migrate/remove_unused_teams.py

```diff
@@ -0,0 +1,116 @@
+    members_but_nothing_else = []
+    team_to_members = {}
+    for page in paged(api.teams.list, org, per_page=100):
+        teams.extend(page)
+
+    for team in teams:
+        # we ignore the fact that there could be multiple pages of responses because we don't need
+        # a full list of teams, since we're just looking for the teams that don't have any of these
+        # objects.
+        repos = api.teams.list_repos_in_org(org, team.slug)
+        projects = api.teams.list_projects_in_org(org, team.slug)
+        child_teams = api.teams.list_child_in_org(org, team.slug)
+
+        # We get all the team members so we can review teams that still
+        # have members.
+        team_members = []
+        for page in paged(api.teams.list_members_in_org, org, team.slug):
+            team_members.extend(page)
+
+        team_to_members[team.slug] = [member.login for member in team_members]
+        if len(repos) == 0 and len(projects) == 0 and len(child_teams) == 0:
+            if len(team_members) == 0:
+                fully_empty_teams.append(team.slug)
+            else:
+                members_but_nothing_else.append(team.slug)
+    return (fully_empty_teams, members_but_nothing_else, team_to_members)
+
+
+@click.option(
+    "--org", default="openedx", help="The github org that you wish to cleanup."
+)
+@click.option(
+    "--dry-run",
+    "-n",
+    default=False,
+    is_flag=True,
+    help="Show what changes would be made without making them.",
+)
+@click.option(
+    "--refresh-cache",
+    help="Refresh cache of github data before running.",
+    default=False,
+    is_flag=True,
+)
+@click.command()
+def main(org, dry_run, refresh_cache):
+    if refresh_cache:
+        load_teams_data_from_github.cache_clear()
+
+    deletion_count = 0
+    api = GhApi()
+    (
+        fully_empty_teams,
+        members_but_nothing_else,
+        team_to_members,
+    ) = load_teams_data_from_github(org)
+
+    click.secho(
+        "The following teams have no related repos, projects, members, or child teams:",
+        bold=True,
+    )
+    click.secho("    - ", nl=False, fg="red")
+    click.secho("\n    - ".join(fully_empty_teams), fg="red")
+
+    if len(fully_empty_teams) > 0 and (
+        dry_run or click.confirm("Delete all fully empty teams?")
+    ):
+        click.secho("Deleting fully empty teams.", bold=True, fg="red")
+        # TODO: Actually do the deletion here.
+        for team in fully_empty_teams:
+            click.secho(f"Deleting '{team}'...", nl=False, fg="red")
+            if not dry_run:
+                api.teams.delete_in_org(org, team)
+            click.secho(f"Done", fg="red")
+            deletion_count += 1
+
+    for team in members_but_nothing_else:
+        click.secho(f"'{team}' has the following members:", bold=True)
+        click.secho("    - ", nl=False)
+        click.secho("\n    - ".join(team_to_members[team]))
+
+        if dry_run or click.confirm(f"Delete '{team}'?"):
```

Comment:
It's okay to run through these individually? I feel like this is the sort of thing I'd be afraid of messing up if I was typing it live with a non-trivial number of teams.

Reply:
yea, I was debating this part, an alternative I considered is that I could turn this part into a separate script that reports the data and one that will delete teams on a given list.  I think it's not too scary but I'm happy to reconsider it.

Reply:
Given that the failure scenario isn't hard to recover from, I agree–not too scary.